lib/multi_configs.js

```diff
@@ -0,0 +1,129 @@
+const fs = require('fs');
+const path = require('path');
+
+const isArray = require('lodash.isarray');
+const isString = require('lodash.isstring');
+const isPlainObject = require('lodash.isplainobject');
+const mapValues = require('lodash.mapvalues');
+
+const expandTilde = require('expand-tilde');
+const parsePath = require('parse-filepath');
+
+
+function search(options, extensions, cwd) {
+  cwd = path.resolve(isString(cwd) ? cwd : '.');
+  extensions = normalizeExtensions(extensions);
+
+  if (!isPlainObject(options)) {
+    return {};
+  }
+
+  return mapValues(options, function(opt, name) {
+    return firstUnNull(expandEachProp(opt, name), findWithExpandedPaths);
```

Comment:
@phated The usage of `lodash.find` is similar to `lodash.filter` and it doesn't return a result by argument function. So if it is substituted for `firstUnNull`, the code would be as follows:

```js
  return (lodash.find(expandEachProp(opt, name), function(expandedPath) {
    return (expandedPath.searchedPath = findWithExpandedPaths(expandedPath)) != null;
  }) || {}).searchedPath;
```

Is this code along your thoughts?